### DISCUSSION TITLE
share an array between files

### DISCUSSION THREAD

**Discussion by aiUser1001** [danny-avila/LibreChat#4227] [General]
I've tried everything I can think of and it still doesn't work. I've created an array in createContextHandlers.js and I exported it module.exports = { myArray, createContextHandlers }. I did the same in the index.js file in the same directory. I would like to read myArray in Markdown.tsx. I did set the following in tsconfig.json:

"allowJs": true,
"outDir": "./dist",

I set the paths in both tsconfig and vite.config. I've tried importing the array in Markdown.tsx, both using the ES and commonJS styles but nothing works, I only get Uncaught ReferenceError: require is not defined. Thanks for any help.

**Comment by danny-avila** [comment]
if you're trying to use a variable in both `client` and `api` workspaces, you need to create it in `packages/data-provider/src` then compile the code (done automatically when you run `npm run frontend`)

**Reply by aiUser1001** [reply]
Thanks for the quick response. I would like to populate myArray in createContextHandlers and use it in Markdown.tsx. Would your answer still work?. I guess I am still confused how to do it.

**Reply by aiUser1001** [reply]
I define the variable along with an event trigger in packages/data-provider/src. I push data to myArray in createContextHandlers but the listener event doesn't fire in Markdown. Still struggling and your help Danny is much appreciated.

**Reply by aiUser1001** [reply]
BTW, I listen for the event in useEffect in Markdown

**Reply by aiUser1001** [reply]
too bad, very convoluted code. Still not able to make it work, I admit that I am not good with TS, JS and React but that consumed way more time than expected. Bummer
